backend/src/main/java/com/ideahub/dao/IdeaPartTypeDAO.java

```diff
@@ -20,4 +23,10 @@ public IdeaPartTypeDAO(final SessionFactory sessionFactory) {
     public List<IdeaPartType> getIdeaDefinition() {
         return this.list(this.criteria());
     }
+
+    public Optional<IdeaPartType> findByName(String aTypeName) {
```

Comment:
I created a IdeaTypeDefinitionCache that you could modify to look this up. It's a singleton that gets populated on first reference, you could add another map in there that maps by name if you like.